internal/controller/ols_app_server_reconciliator_test.go

```diff
@@ -100,21 +120,96 @@ var _ = Describe("App server reconciliator", Ordered, func() {
 			Expect(dep.Annotations[OLSConfigHashKey]).NotTo(Equal(oldHash))
 			Expect(dep.Annotations[OLSConfigHashKey]).NotTo(Equal(oldHash))
 		})
+
+		It("should trigger rolling update of the deployment when changing LLM secret content", func() {
+
+			By("Get the deployment")
+			dep := &appsv1.Deployment{}
+			err := k8sClient.Get(ctx, types.NamespacedName{Name: OLSAppServerDeploymentName, Namespace: OLSNamespaceDefault}, dep)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(dep.Spec.Template.Annotations).NotTo(BeNil())
+			oldHash := dep.Spec.Template.Annotations[OLSProviderCredentialsHashKey]
+			Expect(oldHash).NotTo(BeEmpty())
+
+			By("Delete the provider secret")
+			secretDeletionErr := reconciler.Delete(ctx, secret)
+			Expect(secretDeletionErr).NotTo(HaveOccurred())
```

Comment:
why we need delete the secret in stead of an update here?
in a normal kubernetes cluster a secret mounted as a volume cannot be deleted.

Reply:
Here we are not deleting the secret that is mounted inside the deployment. We trying to test its update scenarios, so for that I have added test cases for both updating the secret content and recreating the secret altogether to see how deployment responds to it.